File: CleanAddisBackOfficeApi/src/main/java/com/iyoa/cleanaddis/model/Category.java
package com.iyoa.cleanaddis.data;



import java.util.Date;

import java.util.UUID;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;


import lombok.Data;
import lombok.RequiredArgsConstructor;

import org.hibernate.annotations.GenericGenerator;


@JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
@Entity
@Table(name = "category")
@Data
@RequiredArgsConstructor
public class Category implements java.io.Serializable{

	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(generator = "UUID")
	@GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
	@Column(name = "uuid", updatable = false, nullable = false)
	private java.util.UUID uuid;

	
	@Column(name = "name", unique = true, nullable = false, length = 100)	
	private String name;
	


	@Column(name = "for_what", nullable = false, length = 65535)	
	private String forWhat;

	
	
	
	
	
	
}
	
	

	

File: app/src/main/java/com/iyoa/cleanaddis/data/teaching/Category.kt
package com.iyoa.cleanaddis.data.teaching

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.*

@Entity(tableName = "category")
data class Category(
    @PrimaryKey
    @ColumnInfo(name="uuid") val uuid : Long,

    @ColumnInfo(name="name") val name:String,
    @ColumnInfo(name="for_what")val for_what: String




) : Serializable{

}
